pallets/delegation/src/migrations/v1.rs

```diff
@@ -97,7 +100,9 @@ fn migrate_roots<T: Config>(new_nodes: &mut BTreeMap<DelegationNodeIdOf<T>, Dele
 			// later, when we know we have seen all the children already.
 			let mut new_root_node = DelegationNode::new_root_node(old_root_id, new_root_details);
 			if let Some(root_children_ids) = deprecated::v1::storage::Children::<T>::take(old_root_id) {
-				new_root_node.children = root_children_ids.iter().copied().collect();
+				let children_set: BTreeSet<DelegationNodeIdOf<T>> = root_children_ids.iter().copied().collect();
```

Comment:
I think this is overkill for now. If the migration fails due to this, it will panic in `migrate_roots` and then we know. In practice, we don't have like any nested delegations on our chains right now so I would like to avoid spending time on something that is not going to matter. What's your take @weichweich .

Reply:
yes i also think this is overkill